src/lib.rs

```diff
@@ -303,7 +326,17 @@ where
         Ok(())
     }
 
-    async fn send_activity_heartbeat(&self, _task_token: ActivityHeartbeat) -> Result<(), ()> {
+    async fn record_activity_heartbeat(
+        &self,
+        _details: ActivityHeartbeat,
+    ) -> Result<RecordActivityHeartbeatResponse, ActivityHeartbeatError> {
+        unimplemented!()
+    }
+
+    async fn get_last_activity_heartbeat(
```

Comment:
What is this for?
When is lang supposed to call this?

Reply:
When you want details of the last heartbeat back, technically you have it, but that way you would need to implement the logic on the lang side. (e.g. latest heartbeat details sent or heartbeat details from the activity task)

Reply:
As discussed, going to ditch this from the core and let lang track heartbeats.